README.md
"# Demon-s-Survivor" 


Dario/Changes.txt
Main.html
Estilos.css
app.js
UnJugador.html

scriptAMA.js
var pjImage = new Image();
pjImage.src = "Spritesheet.png";

var fondoImg = new Image();
fondoImg.src = "Mapa Prueba.jpg";

function sprite (options) {
  var that = {},
      frameIndex = 0,
      tickCount = 0,
      ticksPerFrame = 3,
      numberOfFramesX = options.numberOfFramesX || 9,
      numberOfFramesY = options.numberOfFramesY || 4;
  
  that.context = options.context;
  that.width = options.width;
  that.height = options.height;
  that.xPos = options.xPos;
  that.yPos = options.yPos;
  that.xSpeed = options.xSpeed;
  that.ySpeed = options.ySpeed;
  that.dir = options.dir;
  that.stop = options.stop;
  that.image = options.image;

  that.render = function (){
    that.context.clearRect(0, 0, 600, 400);
    
    that.context.drawImage(
      that.image,
      frameIndex * that.width / numberOfFramesX,    //xPos en la imagen
      that.dir * that.height / numberOfFramesY,   //yPos en la imagen
      that.width / numberOfFramesX,   //width en la imagen
      that.height / numberOfFramesY,  //height en la imagen
      that.xPos,  //xPos en el canvas
      that.yPos,  //yPos en el canvas
      50,    //Animation width
      50);   //Animation height
  };
  
  
  that.update = function () {
    tickCount += 1;
    
    if(tickCount > ticksPerFrame){
      tickCount = 0;
      
      if(frameIndex < numberOfFramesX - 1){
        frameIndex += 1;
      }else{
        frameIndex = 0;
      }
      
    }
    
  };


  
  return that;
}

var canvas = document.getElementById("pjAnimation");
canvas.width = 1152;
canvas.height = 318;

var background = document.getElementById("fondo");
var BG = background.getContext("2d");
BG.drawImage(fondoImg, 1152, 318);

var pj = sprite ({
  context: canvas.getContext("2d"),
  width: 450,
  height: 256,
  xPos: 0,
  yPos: 0,
  xSpeed: 0,
  ySpeed: 0,
  dir: 0,
  stop: true,
  numberOfFramesX: 9,
  numberOfFramesY:4,
  image: pjImage
});

function gameLoop () {
  window.requestAnimationFrame(gameLoop);
  if (pj.stop == false) {
    pj.xPos += pj.xSpeed;
    pj.yPos += pj.ySpeed;
    pj.update();
    pj.render();
  }
}

function move(e){

  if(e.keyCode==39 && pj.xPos<550){
    pj.xSpeed = 3;
    pj.dir = 2;
  }
  if(e.keyCode==37 && pj.xPos>0){
    pj.xSpeed = -3;
    pj.dir = 1;
  }
  if(e.keyCode==40 && pj.yPos<350){
    pj.ySpeed = 3;
    pj.dir = 0
  }
  if(e.keyCode==38 && pj.yPos>0){
    pj.ySpeed = -3;
    pj.dir = 3;
  }

  pj.stop = false;
}

function stop(){
  pj.stop = true;
  pj.frameIndex = 0;
  pj.xSpeed = 0;
  pj.ySpeed = 0;
  pj.render();
}

document.onkeydown = move;
document.onkeyup = stop;

pjImage.addEventListener("load", gameLoop);


Dario/app - copia.js
console.log("Your fucker mother in the shower");
var context;
let game;
var tpad = document.getElementById('controller');
console.log(tpad);
class Personaje {
	
	constructor(x,y,color){
		this.x = x;
		this.y = y;
		this.color = color;
	}
	
	draw(context) {		
		context.fillStyle = this.color;
		context.fillRect(this.x, this.y, 50, 50);		
	}
}

class Game {

	constructor(){
		this.fps = 30;
		this.nextFrame = null;
		this.interval = null;
		this.direction = 'none';
		this.gridSize = 10;
		this.skipTicks = 1000 / this.fps;
		this.nextGameTick = (new Date).getTime();
		this.rect = new Personaje (5,5,"black");
	}

	initialize() {	
	
		var canvas = document.getElementById('playground');

		if (!canvas.getContext) {
			console.log('Error: 2d canvas not supported by this browser.');
			return;
		}
		
		context = canvas.getContext('2d');
		canvas.style.backgroundColor = "red";
		//var rect = new Personaje(20, 20, "black");
		this.rect.draw(context);
		//var rect = context.fillRect(10,10,30,30);
		//document.getElementById("playground").style.backgroundColor = "lightblue";
		window.addEventListener('keydown', e => {
			
			var code = e.keyCode;
			if (code > 36 && code < 41) {
				switch (code) {
				case 37: //flecha izquierda
					this.rect.x -=5;
					break;
				case 38: //flecha abajo
					this.rect.y -=5;
					break;
				case 39: //flecha derecha
					this.rect.x +=5;
					break;
				case 40: //flecha arriba
					this.rect.y +=5;
					break;
				}
			}
		}, false);
		
	}

	setDirection(direction) {
		
		this.direction = direction;
	}

	startGameLoop() {
		
		this.nextFrame = () => {
			requestAnimationFrame(() => this.run());
		}	
		this.nextFrame();		
	}

	stopGameLoop() {
		
		this.nextFrame = null;
		if (this.interval != null) {
			clearInterval(this.interval);
		}
	}

	draw() {
		
		//if(!finalizado){
		context.clearRect(0, 0, 800, 800);
		/*for (var id in this.snakes) {			
			this.snakes[id].draw(this.context);		
		}*/		
		//rect.draw(this.context);	
		this.rect.draw(context);
					
	}

	run() {
	
		while ((new Date).getTime() > this.nextGameTick) {
			this.nextGameTick += this.skipTicks;
			//console.log("Entra en GLOOP");
		}
		
		this.draw();
		if (this.nextFrame != null) {
			this.nextFrame();
		}
	}
	



}

game = new Game();
game.initialize();
game.startGameLoop()

Dario/app.js
console.log("Your fucker mother in the shower");
var context;
let game;
var tpad = document.getElementById('controller'); //Guardamos el elemento
console.log(tpad);
class Personaje {
	
	constructor(x,y,color){
		this.x = x;
		this.y = y;
		this.color = color;
	}
	
	draw(context) {		
		context.fillStyle = this.color;
		context.fillRect(this.x, this.y, 50, 50);		
	}
}

class Game {

	constructor(){
		this.fps = 30;
		this.nextFrame = null;
		this.interval = null;
		this.direction = 'none';
		this.gridSize = 10;
		this.skipTicks = 1000 / this.fps;
		this.nextGameTick = (new Date).getTime();
		this.rect = new Personaje (5,5,"black");
	}

	initialize() {	
	
		var canvas = document.getElementById('playground');

		if (!canvas.getContext) {
			console.log('Error: 2d canvas not supported by this browser.');
			return;
		}
		
		context = canvas.getContext('2d');
		canvas.style.backgroundColor = "red";
		//var rect = new Personaje(20, 20, "black");
		this.rect.draw(context);
		//var rect = context.fillRect(10,10,30,30);
		//document.getElementById("playground").style.backgroundColor = "lightblue";
		tpad.addEventListener('touchstart', e => { //Añadimos los elementos de touch al TPAD.
			console.log('x = ' + e.targetTouches[0].pageX);
			console.log('y = ' + e.targetTouches[0].pageY);
			var x = e.targetTouches[0].pageX;
			var y = e.targetTouches[0].pageY;
			
			console.log(x);
			console.log(y);
			if (x >= 420 && x <= 500){ //Comprobamos los movimientos con coordenadas.
				if (y >= 1038 && y <= 1097){
					this.rect.y -=5;
				}else{
					this.rect.y +=5;
				}
			}else if(x>=362 && x<= 417 ){
				this.rect.x -=5;
			}else if(x>=500 && x<=585){
				this.rect.x +=5;
			}
				
					
				
			var code = e.keyCode;
			if (code > 36 && code < 41) {
				switch (code) {
				case 37: //flecha izquierda
					this.rect.x -=5;
					break;
				case 38: //flecha abajo
					this.rect.y -=5;
					break;
				case 39: //flecha derecha
					this.rect.x +=5;
					break;
				case 40: //flecha arriba
					this.rect.y +=5;
					break;
				}
			}
		}, false);
		
	}

	setDirection(direction) {
		
		this.direction = direction;
	}

	startGameLoop() {
		
		this.nextFrame = () => {
			requestAnimationFrame(() => this.run());
		}	
		this.nextFrame();		
	}

	stopGameLoop() {
		
		this.nextFrame = null;
		if (this.interval != null) {
			clearInterval(this.interval);
		}
	}

	draw() {
		
		//if(!finalizado){
		context.clearRect(0, 0, 800, 800);
		/*for (var id in this.snakes) {			
			this.snakes[id].draw(this.context);		
		}*/		
		//rect.draw(this.context);	
		this.rect.draw(context);
					
	}

	run() {
	
		while ((new Date).getTime() > this.nextGameTick) {
			this.nextGameTick += this.skipTicks;
			//console.log("Entra en GLOOP");
		}
		
		this.draw();
		if (this.nextFrame != null) {
			this.nextFrame();
		}
	}
	



}

game = new Game();
game.initialize();
game.startGameLoop()